src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -498,6 +498,24 @@ bool crop_in_place_optimization::match(const program_node& node,
                 crop_params.input_offsets[0].spatial[1] != 0) {
                 return false;
             }
+            // Do not optimize out for possibly crop padding to select gemm_tiled_opt kernel
+            auto crop_layout = node.get_output_layout();
+            auto& crop_node = node.as<crop>();
+            std::pair<const program_node*, layout> user_info;
+            if (can_crop_be_optimized_simple_data_format(crop_layout, input_layout)) {
+                auto has_padding_expect_batch = [](const padding pad) {
```

Comment:
nit: could you explain why you implemented lambda function instead of just running it?